src/common/entity/domain_icon.ts

```diff
@@ -108,7 +108,7 @@ export const domainIcon = (
         case "switch":
           return compareState === "on" ? mdiToggleSwitch : mdiToggleSwitchOff;
         default:
-          return mdiFlash;
+          return mdiToggleSwitchOutline;
```

Comment:
Well, it is not exactly the same, that's why I suggested the outline variant.

But leaving that aside, you would rather see an icon that mimics a physical switch, regardless of state?

Reply:
that would make more sense.

also looping in @matthiasdebaat

Reply:
ok, I've changed accordingly.
thx

Reply:
Personally, I wouldn't change this since no one has opened a request to do so. It only creates more confusion in the release when people's UI changes as not part of a operational fix. IMO.
Since the customization allows users to override the switch icon, is it really necessary?
This was pushed as a part of an issue I opened regarding the toggle icon override (assumed_state) not work. And this PR does not address that issue.
m2c

Reply:
Tbh to you, this pr has really nothing to do with your issue, though I might have given that impression.

Imho has been long overdue to change an obvious 'incorrect' icon. One that represents multiple and very different things in Ha. Which in itself is the main reason for me suggest this.

Why keep something if we can optimize it so very easily like this?

Hardly a breaking change. Even better: it would prevent people from having to customize the icon in the first place...

A switch is a switch and not a flash, which is power.

I feel this is best seen on an assumed switch where 2 flash using are now shown. Which makes all new HA users post questions in the community. Makes that light-switch and it would be way motie clear what it does.

Reply:
- Before (with flash icon): Users who didn't like it could change it, and did.
- Now (with power icon): Users who liked the flash icon now have to change it back and/or left wondering why their UI is different. Those who already changed it see nothing different.

At the very least this will just create confusion as why this UI changed and perhaps create more needless work for the user to set it back. And no research into what users would expect it to look like. 

I, for one, use the mdi:power icon as it reflects the button on the Sonoff switches.

So, rather than put the work into fixing a newly introduced bug on Feb 3 with #11522 so that users can override an icon as described in the documentation, this is 'fixed' for no apparent reason (IMO). Seems like a waste of time and effort (even if its small). 

That's my 2c. I have nothing more to say

Reply:
just to be clear: this PR doesnt change mdi:flash to mdi:power, it changes mdi:flash to mdi:light-switch.

which is the physical representation of the device, as Paulus suggested.

Reply:
I've suggested a new icon for MDI that resembles a switch. Will check on the status of that.
https://github.com/Templarian/MaterialDesign/issues/6447

Reply:
a that is magic yes. a slimmed down light-switch. great!
what would your suggestion be for a 'default' icon if those could be adapted? 
we do need a variant for when the state is not certain, or not set by the device_class after all.

Reply:
It depends. What do we do now with `light`?

Reply:
test fixed: https://github.com/home-assistant/frontend/blob/8301ae262cdbb1402a123d527fc6aeef422ef6a6/src/common/const.ts#L36